pkg/controller/util.go

```diff
@@ -28,36 +29,59 @@ func (c *Controller) makeClusterConfig() cluster.Config {
 	}
 }
 
-func thirdPartyResource(TPRName string) *extv1beta.ThirdPartyResource {
-	return &extv1beta.ThirdPartyResource{
-		ObjectMeta: metav1.ObjectMeta{
-			//ThirdPartyResources are cluster-wide
-			Name: TPRName,
-		},
-		Versions: []extv1beta.APIVersion{
-			{Name: constants.TPRApiVersion},
-		},
-		Description: constants.TPRDescription,
-	}
-}
-
 func (c *Controller) clusterWorkerID(clusterName spec.NamespacedName) uint32 {
 	return crc32.ChecksumIEEE([]byte(clusterName.String())) % c.opConfig.Workers
 }
 
-func (c *Controller) createTPR() error {
-	tpr := thirdPartyResource(constants.TPRName)
+func (c *Controller) createCRD() error {
+	crd := &apiextv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: constants.CRDResource + "." + constants.CRDGroup,
```

Comment:
Do we need to specify the namespace here as well (since we are using the NamespacedScope)?

Reply:
makes sense

Reply:
actually no, CRD is cluster wide